src/System.Text.Json/tests/JsonObjectTests.cs

```diff
@@ -712,11 +654,147 @@ public static void TestArgumentNullValidation()
         }
 
         [Fact]
-        public static void TestDuplicatesEnumOutOfRange()
+        public static void TestStringComparisonEnum()
+        {
+            var jsonObject = new JsonObject()
+            {
+                { "not encyclopaedia", "value1" },
+                { "Encyclopaedia", "value2" },
+                { "NOT encyclopaedia", "value3" },
+                { "encyclopaedia", "value4" }
+            };
+
+            Assert.Equal(4, jsonObject.Count());
+
+            Assert.False(jsonObject.ContainsProperty("ENCYCLOPAEDIA"));
+            Assert.False(jsonObject.TryGetPropertyValue("ENCYCLOPAEDIA", out JsonNode jsonNode));
+            Assert.Null(jsonNode);
+            Assert.Throws<KeyNotFoundException>(() => jsonObject.GetPropertyValue("ENCYCLOPAEDIA"));
+            jsonObject.Remove("ENCYCLOPAEDIA");
+            Assert.Equal(4, jsonObject.Count());
+
+            Assert.False(jsonObject.ContainsProperty("ENCYCLOPAEDIA", StringComparison.CurrentCulture));
```

Comment:
We should add tests where we set the current culture to something else and then check both cases (using `StringComparison.CurrentCulture` finds what we are looking for because now it matches, and one where because we use currentculture, it doesn't).